buffer/src/main/java/io/netty/buffer/AdvancedLeakAwareByteBuf.java

```diff
@@ -53,7 +53,7 @@
         this.leak = leak;
     }
 
-    private void recordLeakNonRefCountingOperation() {
+    static void recordLeakNonRefCountingOperation(ResourceLeak leak) {
```

Comment:
Since you're sharing this now, would it make sense to move this to some sort of common util class? It seems to be the only thing that uses `ACQUIRE_AND_RELEASE_ONLY`, so that could move too.